sql/statesql/migrations/state_0021_migration_test.go

```diff
@@ -13,44 +13,21 @@ import (
 	"github.com/spacemeshos/go-spacemesh/signing"
 	"github.com/spacemeshos/go-spacemesh/sql"
 	"github.com/spacemeshos/go-spacemesh/sql/atxs"
+	"github.com/spacemeshos/go-spacemesh/sql/statesql"
 )
 
-// Test that in-code migration results in the same schema as the .sql one.
-func Test0021Migration_CompatibleSchema(t *testing.T) {
```

Comment:
I actually added this test to the current version of this PR and it has passed.
The actual reason why it is not necessary is this:
- as of now, `schemagen` command which is used to generate `schema.sql` files for local and state databases imports and uses coded migrations
- `TestIdempotentMigration` in `sql/statesql` and `sql/localsql` uses SQL scripts ("placeholders") instead of coded migrations as it doesn't import them to avoid cyclic dependencies. If the coded migrations and SQL scripts result in different schemas, `TestIdempotentMigration` will fail

Reply:
I think the reason as to why it failed was because of formatting of sql files on my end and yes `TestIdempotentMigration` would also have caught the issue. On that note see my other comment here: https://github.com/spacemeshos/go-spacemesh/pull/6003/files#r1723235468